delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceCommand.scala

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resources
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import io.circe.Json
+
+/**
+  * Enumeration of resource commands
+  */
+sealed trait ResourceCommand extends Product with Serializable {
+
+  /**
+    * @return the project where the resource belongs to
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the resource identifier
+    */
+  def id: Iri
+
+  /**
+    * @return the identity associated to this command
+    */
+  def subject: Subject
+
+}
+
+object ResourceCommand {
+
+  /**
+    * Command that signals the intent to create a new resource.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param schema      the schema used to constrain the resource
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param subject     the subject which created this event
+    */
+  final case class CreateResource(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      source: Json,
+      compacted: CompactedJsonLd,
+      expanded: ExpandedJsonLd,
+      subject: Subject
+  ) extends ResourceCommand
+
+  /**
+    * Command that signals the intent to update an existing resource.
+    *
+    * @param id        the resource identifier
+    * @param project   the project where the resource belongs
```

Comment:
We don't need the project here as a resource can't switch from a project to another right ?

Reply:
The project field is always passed along, since it is the way to generate the persistenceId (which is going to be projectRef + id)

Reply:
Then you have it in the state, no need to pass it again though the command

Reply:
It is needed, you will see it more clear when you have the dummy or Impl

Reply:
What I meant is you can take the projectRef of the state to create the event.